assembly/src/assembler/mod.rs

```diff
@@ -421,6 +421,8 @@ impl Assembler {
                 // in `module_indices` are in AST form by definition.
                 let ast_module = self.linker[module_idx].unwrap_ast().clone();
 
+                mast_forest_builder.merge_advice_map(&ast_module.advice_map)?;
```

Comment:
Question: would advice data from statically linked modules (e.g., the ones added via `compile_and_statically_link()` method) make it into the `MastForestBuilder` as well?

Reply:
It does already when the `MastForestBuilder` is instantiated